R/zchunk_L241.fgas.R

```diff
@@ -33,80 +32,239 @@ module_emissions_L241.fgas_DISABLED <- function(command, ...) {
 
     # Load required inputs
     GCAM_region_names <- get_data(all_data, "common/GCAM_region_names")
-    A_regions <- get_data(all_data, "emissions/A_regions")
-    FUT_EMISS_GV <- get_data(all_data, "emissions/FUT_EMISS_GV")
-    L141.hfc_R_S_T_Yh <- get_data(all_data, "L141.hfc_R_S_T_Yh")
+    A_regions         <- get_data(all_data, "emissions/A_regions")
+    FUT_EMISS_GV      <- get_data(all_data, "emissions/FUT_EMISS_GV")
     L142.pfc_R_S_T_Yh <- get_data(all_data, "L142.pfc_R_S_T_Yh")
-    L141.hfc_ef_R_cooling_Yh <- get_data(all_data, "L141.hfc_ef_R_cooling_Yh")
+
+    get_data(all_data, "temp-data-inject/L141.hfc_R_S_T_Yh")  %>%
+      # The following 2 lines of code will be removed latter when we are using the 'real' data
+      gather(year, value, -GCAM_region_ID, -supplysector, -subsector, -stub.technology, -Non.CO2) %>%
+      mutate(year = as.numeric(substr(year, 2, 5))) -> L141.hfc_R_S_T_Yh
+
+    get_data(all_data, "temp-data-inject/L141.hfc_ef_R_cooling_Yh") %>%
+      # The following 2 lines of code will be removed latter when we are using the 'real' data
+      gather(year, value, -GCAM_region_ID, -supplysector, -subsector, -stub.technology, -Non.CO2) %>%
+      mutate(year = as.numeric(substr(year, 2, 5))) -> L141.hfc_ef_R_cooling_Yh
+
 
     # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
+    # Format and round emission values for HFC gas emissions for technologies in all regions.
+    L141.hfc_R_S_T_Yh %>%
+      filter(year %in% BASE_YEARS) %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      mutate(input.emissions = round(value, emissions.DIGITS_EMISSIONS)) %>%
+      select(-GCAM_region_ID, -value) ->
+      L241.hfc_all
+
+
+    # L241.pfc: F-gas emissions for technologies in all regions.
     #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
+    # Remove anything that's zero in all base years for any technology, because no future
+    # coefs are read in for any techs.
+    L142.pfc_R_S_T_Yh %>%
+      filter(year %in% emissions.HISTORICAL) %>%
+      spread(year, value) %>%
+      mutate(yrsum = rowSums(.[grep(YEAR_PATTERN, names(.))])) %>%
+      filter(yrsum != 0) %>%
```

Comment:
I tried several variations of `group_by(year)` %>% filter(sum(value) != 0)` but could not filter out the correct number of rows.